doc/source/walkthrough.rst

```diff
@@ -0,0 +1,311 @@
+Walkthrough
+===========
+
+This walkthrough will overview the core concepts of Ray:
+
+   1. Using remote functions (tasks) [``ray.remote``]
+   2. Fetching results (object IDs) [``ray.put``, ``ray.get``, ``ray.wait``]
+   3. Using remote classes (actors) [``ray.remote``]
+
+.. list-table:: Title
+   :widths: 25 25 50
+   :header-rows: 1
+
+   * - Heading row 1, column 1
+     - Heading row 1, column 2
+     - Heading row 1, column 3
+   * - Row 1, column 1
+     -
+     - Row 1, column 3
+   * - Row 2, column 1
+     - Row 2, column 2
+     - Row 2, column 3
+
+
+With Ray, your code will work on a single machine and can be easily scaled to a
+large cluster. To run this walkthrough, install Ray with ``pip install -U ray``.
+
+.. code-block:: python
+
+  import ray
+
+  # Start Ray. If you're connecting to an existing cluster, you would use
+  # ray.init(redis_address=<cluster-redis-address>) instead.
+  ray.init()
+
+See the `Configuration <configure.html>`__ documentation for the various ways to
+configure Ray. To start a multi-node Ray cluster, see the `cluster setup page
+<using-ray-on-a-cluster.html>`__. You can stop ray by calling
+``ray.shutdown()``. To check if Ray is initialized, you can call
+``ray.is_initialized()``.
+
+Remote functions (Tasks)
+------------------------
+
+Ray enables arbitrary Python functions to be executed asynchronously. These asynchronous Ray functions are called "remote functions". The standard way to turn a Python function into a remote function is to add the ``@ray.remote`` decorator. Here is an example.
+
+.. code:: python
+
+    # A regular Python function.
+    def regular_function():
+        return 1
+
+    # A Ray remote function.
+    @ray.remote
+    def remote_function():
+        return 1
+
+This causes a few things changes in behavior:
+
+    1. **Invocation:** The regular version is called with ``regular_function()``, whereas the remote version is called with ``remote_function.remote()``.
+    2. **Return values:** ``regular_function`` immediately executes and returns ``1``, whereas ``remote_function`` immediately returns an object ID (a future) and then creates a task that will be executed on a worker process. The result can be retrieved with ``ray.get``.
+
+    .. code:: python
```

Comment:
I'd like to push back on this - I think this is non-standard and not as reader friendly as you expect, especially for those who are glancing over documentation.

For examples, this is definitely necessary. 

On the other hand, I can leave a note to show that it's actually possible to copy and paste these: https://numpydoc.readthedocs.io/en/latest/format.html#sections

In the future, we can incorporate a modded `copybutton`: https://docs.scipy.org/doc/numpy-1.16.0/reference/generated/numpy.random.randint.html
https://github.com/scipy/scipy-sphinx-theme/blob/c466764e2231ba132c09826b5b138fffa1cfcec3/_theme/scipy/static/js/copybutton.js

Reply:
The reason why it's not reader friendly is because the in-line comment is not a standard signifier for return value. However `>>>` and subsequent lines are standard for python interpreter output.